lib/sycamore/sycamore/writers/duckdb_writer.py

```diff
@@ -0,0 +1,145 @@
+from dataclasses import dataclass, asdict, is_dataclass
+from typing import Optional, Any
+from typing_extensions import TypeGuard
+
+from sycamore.data.document import Document
+from sycamore.writers.base import BaseDBWriter
+from sycamore.writers.common import convert_to_str_dict
+import pyarrow as pa
+import duckdb
+
+
+@dataclass
+class DuckDBClientParams(BaseDBWriter.ClientParams):
+    pass
+
+
+@dataclass
+class DuckDBTargetParams(BaseDBWriter.TargetParams):
+    db_url: Optional[str] = ":default:"
+    table_name: Optional[str] = "default_table"
+
+    def compatible_with(self, other: BaseDBWriter.TargetParams) -> bool:
+        if not isinstance(other, DuckDBTargetParams):
+            return False
+        if self.db_url != other.db_url:
+            return False
+        if self.table_name != other.table_name:
+            return False
+        return True
+
+
+class DuckDBClient(BaseDBWriter.Client):
+    def __init__(self, client_params: DuckDBClientParams):
+        pass
+
+    @classmethod
+    def from_client_params(cls, params: BaseDBWriter.ClientParams) -> "DuckDBClient":
+        assert isinstance(params, DuckDBClientParams)
+        return DuckDBClient(params)
+
+    def write_many_records(
+        self, records: list[BaseDBWriter.Record], target_params: BaseDBWriter.TargetParams, batch_size: int = 1000
+    ):
+        N = 1000 ^ 2  # Bit lesser than 1 MB
```

Comment:
1000 ^ 2 = 1002 apparently. You want 1000 ** 2. (Or if you're ok with exactly 1 MB 2 ** 20)... or just write out the number (python also accepts _'s in numbers so you could say 1_000_000)